core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -122,7 +126,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
         } else {
           logInfo("Registered executor: " + executorRef + " with ID " + executorId)
           context.reply(RegisteredExecutor)
-
+          latestTokens.foreach(x => context.reply(NewTokens(x)))
```

Comment:
Yep, I understand that. But that issue exists for pretty much all messages sent within Spark - it is not clear whether we do anything at all to handle retries outside of what akka does (if any). Anyway, I am going to try once more - if test fail, I will just stick to 2 messages rather than risk random regressions